pal/input/native/gamepad-input.ts

```diff
@@ -1,7 +1,374 @@
+        this._buttonOptions = new InputSourceButton();
+        this._buttonOptions.getValue = () => this._nativeButtonState[Button.NS_PLUS];  // TODO: NX only for now
+
+        const dpadUp = new InputSourceButton();
+        dpadUp.getValue = () => this._nativeButtonState[Button.DPAD_UP];
+        const dpadDown = new InputSourceButton();
+        dpadDown.getValue = () => this._nativeButtonState[Button.DPAD_DOWN];
+        const dpadLeft = new InputSourceButton();
+        dpadLeft.getValue = () => this._nativeButtonState[Button.DPAD_LEFT];
+        const dpadRight = new InputSourceButton();
+        dpadRight.getValue = () => this._nativeButtonState[Button.DPAD_RIGHT];
+        this._dpad = new InputSourceDpad({ up: dpadUp, down: dpadDown, left: dpadLeft, right: dpadRight });
+
+        this._buttonLeftStick = new InputSourceButton();
+        this._buttonLeftStick.getValue = () => this._nativeButtonState[Button.BUTTON_LEFT_STICK];
+        const leftStickUp = new InputSourceButton();
+        leftStickUp.getValue = () => this._nativeButtonState[Button.LEFT_STICK_UP];
+        const leftStickDown = new InputSourceButton();
+        leftStickDown.getValue = () => this._nativeButtonState[Button.LEFT_STICK_DOWN];
+        const leftStickLeft = new InputSourceButton();
+        leftStickLeft.getValue = () => this._nativeButtonState[Button.LEFT_STICK_LEFT];
+        const leftStickRight = new InputSourceButton();
+        leftStickRight.getValue = () => this._nativeButtonState[Button.LEFT_STICK_RIGHT];
+        this._leftStick = new InputSourceStick({ up: leftStickUp, down: leftStickDown, left: leftStickLeft, right: leftStickRight });
+
+        this._buttonRightStick = new InputSourceButton();
+        this._buttonRightStick.getValue = () => this._nativeButtonState[Button.BUTTON_RIGHT_STICK];
+        const rightStickUp = new InputSourceButton();
+        rightStickUp.getValue = () => this._nativeButtonState[Button.RIGHT_STICK_UP];
+        const rightStickDown = new InputSourceButton();
+        rightStickDown.getValue = () => this._nativeButtonState[Button.RIGHT_STICK_DOWN];
+        const rightStickLeft = new InputSourceButton();
+        rightStickLeft.getValue = () => this._nativeButtonState[Button.RIGHT_STICK_LEFT];
+        const rightStickRight = new InputSourceButton();
+        rightStickRight.getValue = () => this._nativeButtonState[Button.RIGHT_STICK_RIGHT];
+        this._rightStick = new InputSourceStick({ up: rightStickUp, down: rightStickDown, left: rightStickLeft, right: rightStickRight });
+
+        this._buttonMenu = new InputSourceButton();
+        this._buttonMenu.getValue = () => this._nativeButtonState[Button.ROKID_MENU];  // TODO: Rokid only for now
+        this._buttonStart = new InputSourceButton();
+        this._buttonStart.getValue = () => this._nativeButtonState[Button.ROKID_START];  // TODO: Rokid only for now
```

Comment:
Is the buttonStart for share function ?
can we merge `Button.ROKID_START` and `Button.NS_MINUS` ?